### PR TITLE
[BACKWARDS INCOMPATIBLE] Redo migrations

### PR DESCRIPTION
**Pull request by jbrooksuk**
I've rewritten all existing migrations to merge into one migration per table. I have however removed the web hook tables and models, we'll come back to that properly when we're ready to work on them.

**Please note that all existing installations will be unable to migrate past this point. This is a backwards incompatible change.**

We've been experiencing issues when upgrading PgSQL databases which prevent components from being added without groups. This change will fix the migrations.

### CODE REVIEW

File: app/database/migrations/2015_01_05_203014_CreateSettingsTable.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ public function up()
         Schema::create('settings', function (Blueprint $table) {
             $table->increments('id');
             $table->string('name');
-            $table->text('value')->nullable()->default(null);
+            $table->longText('value');
```
Comment (joecohens) [inline app/database/migrations/2015_01_05_203014_CreateSettingsTable.php]: Should this have a default?

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203014_CreateSettingsTable.php]: Nah, settings will always have a value, won't they?

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203014_CreateSettingsTable.php]: We shouldn't abuse default values.

Comment (joecohens) [inline app/database/migrations/2015_01_05_203014_CreateSettingsTable.php]: Ok ok. :)

**Comment by joecohens** [comment]
:+1:

**Comment by jbrooksuk** [comment]
Rebased.

**Comment by jbrooksuk** [comment]
@GrahamCampbell ok to merge?

File: app/database/migrations/2015_01_05_203341_CreateUsersTable.php [hunk 2]
```diff
@@ -15,10 +15,11 @@ public function up()
     {
         Schema::create('users', function (Blueprint $table) {
             $table->increments('id');
-            $table->string('username', 200)->nullable(false);
-            $table->string('password')->nullable(false);
-            $table->rememberToken();
-            $table->string('email')->nullable()->default(null);
+            $table->string('username');
+            $table->string('password');
+            $table->string('remember_token')->default('');
```
Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: this should be nullable

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: Still with a default of `''`?

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: also should be unique

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: we don't want empty string when there is no token, we surely want null?

File: app/database/migrations/2015_01_05_203341_CreateUsersTable.php [hunk 3]
```diff
@@ -15,10 +15,11 @@ public function up()
     {
         Schema::create('users', function (Blueprint $table) {
             $table->increments('id');
-            $table->string('username', 200)->nullable(false);
-            $table->string('password')->nullable(false);
-            $table->rememberToken();
-            $table->string('email')->nullable()->default(null);
+            $table->string('username');
```
Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: should be unique

File: app/database/migrations/2015_01_05_203341_CreateUsersTable.php (hunk 2 above)
Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: Good point.

File: app/database/migrations/2015_01_05_203341_CreateUsersTable.php (hunk 3 above)
Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: We want email to be unique as well then since that's what we login with.

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: And api_key

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: yeh

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: or better, an index

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: rather than just unique

Comment (joecohens) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: :+1:

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: A unique index?

Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: indexes have to be unique don't they?

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: http://laravel.com/docs/4.2/schema#adding-indexes

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: Done. Check now.

File: app/database/migrations/2015_01_05_203341_CreateUsersTable.php [hunk 4]
```diff
@@ -15,10 +15,11 @@ public function up()
     {
         Schema::create('users', function (Blueprint $table) {
             $table->increments('id');
-            $table->string('username', 200)->nullable(false);
-            $table->string('password')->nullable(false);
-            $table->rememberToken();
-            $table->string('email')->nullable()->default(null);
+            $table->string('username')->unique();
```
Comment (GrahamCampbell) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: we should just add this an "index" shouldn't we?

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: Index the username as `unique`?

Comment (jbrooksuk) [inline app/database/migrations/2015_01_05_203341_CreateUsersTable.php]: That's fine if it's the case, so I assume it's the same with the rememberToken().

**Comment by jbrooksuk** [comment]
@GrahamCampbell ok, I've switched back to `rememberToken` I forgot about that.

**Comment by GrahamCampbell** [comment]
This looks ok. We can always just edit the migrations at a later point and cause more rage, lol.

**Comment by jbrooksuk** [comment]
Prepare your bunkers lads, we're merging.

**Comment by GrahamCampbell** [comment]
lol, i get plenty of abuse whenever i close more than 10 issues to do with eloquent over on laravel/framework anyway. I closed 100 the other week. Big mess, lol.

**Comment by jbrooksuk** [comment]
:hammer:

**Comment by joecohens** [comment]
Haha :)

**Pull request merged**